Users/beaub/scan fix

beaubelgrave opened the pull request:
Fix unit test rename/move issue after rebase and new unsafe precondition checks as part of Rust 1.78.

michaelcfanning commented on ReleaseHistory.md:
```diff
@@ -0,0 +1,10 @@
+# SecurityUtilities Release History
+
+## Unreleased
+* Breaking** Renamed IdentifiableSecrets.GenerateIdentifiableKey to IdentifiableSecrets.GenerateBase64Key.
```
>** [](http://example.com/codeflow?start=10&length=2)

what is this double asterisk for? is there something I don't know about markdown?? :)

marmegh replied:
We've been a little inconsistent with noting breaking changes. Thought I'd try that one out to see what it looks like or if there was a reason. Looks more like a previous typo got reused a few times on breaking changes. Switched to a more uniform approach.

michaelcfanning reviewed (approved):
:shipit:

beaubelgrave committed:
Rust/His: Fix test_file location after rebase

Directories got renamed, but the test_files were missed during that.

Move test_files back under the new rust specific directory.

Signed-off-by: Beau Belgrave <beaub@linux.microsoft.com>

beaubelgrave committed:
Rust/His: Remove unsafe utf16 conversion

UTF16 conversion can happen at any byte offset within data. This cannot
be guaranteed to be aligned. We cannot use unsafe and
slice.from_raw_parts() due to this.

Convert by simply walking bytes and checking most significant byte for
non-zero instead of > 255 on a u16.

Signed-off-by: Beau Belgrave <beaub@linux.microsoft.com>
